lumispy/utils/analysis.py

```diff
@@ -0,0 +1,117 @@
+# -*- coding: utf-8 -*-
+# Copyright 2019-2022 The LumiSpy developers
+#
+# This file is part of LumiSpy.
+#
+# LumiSpy is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the license, or
+# (at your option) any later version.
+#
+# LumiSpy is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with LumiSpy. If not, see <https://www.gnu.org/licenses/#GPL>.
+
+import hyperspy.api as hs
+
+
+def plot_span_map(sig, nspans=1):
+    """
+    Plot a span map.
+    
+    Plots a navigator consisting of the sum over all positions in sig, with `nspans` spans. 
+
+    For each span a `BaseSignal` image is plotted consisting of integrated counts over the the range defined by the span.
+
+    The spans can be moved interactively and the corresponding images will update automatically.
+
+    Arguments
+    ---------
+    sig: hyperspy.signals.Signal1D
+        hyperspectra to inspect. Should have 2 spatial dimensions and 1 signal dimension.
+    nspans: int
+        number of channels to plot. Maximum is 3, default is 1.
+
+    Returns
+    -------
+    all_sum: hs.signals.BaseSignal
+        Sum over all positions of `sig`, the 'navigator' for `plot_span_map`
+    spans: hs.roi.SpanROI
+        The span ROIs from the navigator
+    span_sigs: hs.signals.BaseSignal
+        slices of `sig` according to each span roi
+    span_sums: hs.signals.BaseSignal
+        the summed `span_sigs`.
```

Comment:
Can you also add as argument the key `ax` to select to plot the resulting map to an already existing matplotlib `figure` object? See the plotting hyperspy functions for reference. I think that is very useful, especially when you want to edit all the figure/axis parameters yourself (e.g. imagine I want the imshow plot to be in log scale instead...)

Reply:
I can see why this might be useful, and I think it could be added, however I think this is still possible by using the signals returned by the function.

You can perform the slicing using the `plot_span_map` method and then use the returned signal objects to plot things over whichever axes you would like.

I'd argue it could get quite confusing as to which ax is which etc. and we could end up with many arguments.

Not sure what the consensus is, especially if this is the convention.

This has made me realise that we should probably provide a `spans` argument which allows a user to provide a list of `SpanROI`s to use to slice your data, with ready defined ranges. This is the sort of thing I do often - use interactive ROIs to explore and find a desired ROI, and then overwrite the previous ROI definition to make the default position the one I just found... if that makes sense!

Reply:
OK. I understand. The user would need to replot each component independently... For the ax argument, I think it should return the banpass filtered image/linescan (not the SpanROIs). I would suggest to still allow to set this. But if it is too complicated, then don't worry.

As for the list of SpanROI as arguments, that would be great to be added!

Reply:
> This has made me realise that we should probably provide a spans argument which allows a user to provide a list of SpanROIs to use to slice your data, with ready defined ranges. This is the sort of thing I do often - use interactive ROIs to explore and find a desired ROI, and then overwrite the previous ROI definition to make the default position the one I just found... if that makes sense!

This! I think it's very important also for sharing notebooks etc. Personally I would be in favor of adding a list of tuples, something like `spans=[(beg0,end0),(beg1,end1)]` and not already defined roi objects, as the target audience of this feature, if I am not mistaken, are primarily people who do not want to bother learning how roi objects work am I right?